### PR TITLE
feat: rename aliases to registryAliases

### LINKED ISSUE #15866: Refactor: rename aliases to registryAliases
### Describe the proposed change(s).

Rename existing option `aliases` to be `registryAliases`. Needs migration code added too.

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

- Rename `aliases` -> `registryAliases`
- Only upadted files relate to `helm*` and which are directly using `aliases`
- Migration handled

## Context

- Closes: #15866 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required
Maybe we need to add more info to alert users about the change unless the bot does it automatically.

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
Need to resolve docs issue. WIP

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -795,8 +795,8 @@ const options: RenovateOptions[] = [
     globalOnly: true,
   },
   {
-    name: 'aliases',
-    description: 'Aliases for registries, package manager specific.',
+    name: 'registryAliases',
+    description: 'registryAliases for registries, package manager specific.',
```
Comment (rarkins) [inline lib/config/options/index.ts:799]:
```suggestion
    description: 'Aliases for registries.',
```

File: lib/modules/manager/helm-requirements/readme.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Renovate supports updating Helm Chart references within `requirements.yaml` files.
 
-If your Helm charts make use of repository Aliases then you will need to configure an `aliases` object in your config to tell Renovate where to look for them.
+If your Helm charts make use of repository registryAliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```
Comment (rarkins) [inline lib/modules/manager/helm-requirements/readme.md:3]:
```suggestion
If your Helm charts make use of repository aliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```

File: lib/modules/manager/helmfile/readme.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 Checks `helmfile.yaml` files and extracts dependencies for the `helm` datasource.
 
-If your Helm charts make use of repository Aliases then you will need to configure an `aliases` object in your config to tell Renovate where to look for them.
+If your Helm charts make use of repository registryAliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```
Comment (rarkins) [inline lib/modules/manager/helmfile/readme.md:3]:
```suggestion
If your Helm charts make use of repository aliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```

File: lib/modules/manager/helmv3/readme.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Renovate supports updating Helm Chart references within `requirements.yaml` (Helm v2) and `Chart.yaml` (Helm v3) files.
 
-If your Helm charts make use of repository Aliases then you will need to configure an `aliases` object in your config to tell Renovate where to look for them.
+If your Helm charts make use of repository registryAliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```
Comment (rarkins) [inline lib/modules/manager/helmv3/readme.md:3]:
```suggestion
If your Helm charts make use of repository aliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them.
```

File: lib/modules/manager/types.ts [hunk 5]
```diff
@@ -15,7 +15,9 @@ export interface ManagerData<T> {
 }
 
 export interface ExtractConfig {
-  aliases?: Record<string, string>;
+  registryUrls?: string[];
+  endpoint?: string;
```
Comment (rarkins) [inline lib/modules/manager/types.ts:19]:
```suggestion
```

Seems wrong/accidental addition

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — COMMENTED** [review]
This might be a breaking change if we don't migrate CLI and ENV

**Comment by RahulGautamSingh** [comment]
For CLI and Env, will updates to these two files suffice?

https://github.com/renovatebot/renovate/blob/main/lib/workers/global/config/parse/cli.ts#L19-L35

https://github.com/renovatebot/renovate/blob/4b6fc85146b6563f2da1633518c8c9c54afc2601/lib/workers/global/config/parse/env.ts#L37-L40

**Comment by RahulGautamSingh** [comment]
> For CLI and Env, will updates to these two files suffice?
> 
> https://github.com/renovatebot/renovate/blob/4b6fc85146b6563f2da1633518c8c9c54afc2601/lib/workers/global/config/parse/env.ts#L37-L40

https://github.com/renovatebot/renovate/blob/main/lib/workers/global/config/parse/cli.ts#L19-L35

**Comment by rarkins** [comment]
Yes, I think so. Please test both

File: lib/modules/manager/types.ts (hunk 5 above)
Comment (RahulGautamSingh) [reply lib/modules/manager/types.ts]: I think I meant it to be added in `RenovateConfig`.

**Comment by RahulGautamSingh** [comment]
Tested both locally, [here](https://gist.github.com/RahulGautamSingh/3ddb8520ee2ca2d9db7660e10e0008d0) are the results.

Having problem with ENV formatting I guess. I have added it like this:
image
It gets detected but it is not being used.

File: lib/config/validation.ts [hunk 6]
```diff
@@ -534,7 +534,9 @@ export async function validateConfig(
           currentPath !== 'force.constraints'
         ) {
           if (is.plainObject(val)) {
-            if (key === 'aliases') {
+            if (key === 'registryAliases') {
+              //eslint-disable-next-line
+              console.log('lmao', config.registryAliases);
```
Comment (rarkins) [inline lib/config/validation.ts:539]:
```suggestion
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Multiple console statements, not ready for review

File: lib/workers/global/index.ts [hunk 7]
```diff
@@ -39,6 +39,7 @@ export async function getRepositoryConfig(
   );
   await fs.ensureDir(repoConfig.localDir);
   delete repoConfig.baseDir;
+
```
Comment (rarkins) [inline lib/workers/global/index.ts:42]: revert

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 8]
```diff
@@ -2212,6 +2195,23 @@ If the `versioning` field is missing, then Renovate defaults to using `semver` v
 For more details and examples, see our [documentation for the `regex` manager](/modules/manager/regex/).
 For template fields, use the triple brace `{{{ }}}` notation to avoid Handlebars escaping any special characters.
 
+## registryAliases
+
+The `registryAliases` object is used for configuring registry registryAliases.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2200]:
```suggestion
You can use the `registryAliases` object to set registry aliases.
```

File: docs/usage/configuration-options.md [hunk 9]
```diff
@@ -2212,6 +2195,23 @@ If the `versioning` field is missing, then Renovate defaults to using `semver` v
...
+Currently it is needed/supported for the `helm-requirements`, `helmv3` and `helmfile` managers only.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2201]:
```suggestion
This feature only works with these managers:

- `helm-requirements`
- `helmv3`
- `helmfile`
```

File: docs/usage/configuration-options.md [hunk 10]
```diff
@@ -2212,6 +2195,23 @@ If the `versioning` field is missing, then Renovate defaults to using `semver` v
...
+
+The above managers include this default registryAlias:
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2203]:
```suggestion
The managers listed above all have this default registryAlias:
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
How about something like this?

**Review by HonkingGoose — DISMISSED** [review]

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/helmv3/extract.ts [hunk 11]
```diff
@@ -60,9 +60,8 @@ export async function extractPackageFile(
       res.skipReason = 'no-repository';
       return res;
     }
-
```
Comment (viceice) [inline lib/modules/manager/helmv3/extract.ts:63]: revert this

File: lib/modules/manager/helmv3/utils.ts [hunk 12]
```diff
@@ -8,7 +8,6 @@ export function parseRepository(
   repositoryURL: string
 ): PackageDependency {
   const res: PackageDependency = {};
-
```
Comment (viceice) [inline lib/modules/manager/helmv3/utils.ts:11]: revert this

**Review by viceice — COMMENTED** [review]
some small things

**Review by rarkins — APPROVED** [review]

**Pull request merged**